osu.Game/Screens/SelectV2/ISongSelect.cs

```diff
@@ -16,39 +18,19 @@ public interface ISongSelect
         /// </summary>
         void Delete(BeatmapSetInfo beatmapBeatmapSetInfo);
 
-        /// <summary>
-        /// Requests the user for confirmation to clear all local scores in the given beatmap.
-        /// </summary>
-        void ClearScores(BeatmapInfo beatmap);
-
-        /// <summary>
-        /// Opens beatmap editor with the given beatmap.
-        /// </summary>
-        void Edit(BeatmapInfo beatmap);
-
-        /// <summary>
-        /// Whether calls to <see cref="Edit"/> will succeed or not.
-        /// </summary>
-        bool EditingAllowed { get; }
-
         /// <summary>
         /// Opens the manage collections dialog.
         /// </summary>
         void ManageCollections();
 
         /// <summary>
-        /// Marks a beatmap manually as being played.
-        /// </summary>
-        void MarkPlayed(BeatmapInfo beatmap);
-
-        /// <summary>
-        /// Hides a beatmap from user's vision.
+        /// Present the provided score at the results screen.
         /// </summary>
-        void Hide(BeatmapInfo beatmap);
+        void PresentScore(ScoreInfo score);
 
         /// <summary>
-        /// Present the provided score at the results screen.
+        /// Gets relevant actionable items for beatmap context menus, based on the type of song select.
         /// </summary>
-        void PresentScore(ScoreInfo score);
+        IEnumerable<OsuMenuItem> GetForwardActions(BeatmapInfo beatmap);
```

Comment:
I don't like this. It is not immediately clear to me that this abstraction will scale across however many variants of song select there will need to exist to replace the current one. I would honestly probably take code duplication over this.

I can understand the want to "standardise" actions & order across beatmap options & context menu, but I am not convinced coupling them at *code* level isn't going to backfire a few PRs down the line.

Once all of the weird subclasses are replaced and it is more clear to me that this won't get weird warts because subclass XYZ of song select needs something specific done for it and some sort of `GetSpecialActionsOnlyForThisScreen()` contraption won't have to be conjured, *then* I would have a much easier time being sold on such a change.

Reply:
> Once all of the weird subclasses are replaced and it is more clear to me

If you mean song select subclasses, I already checked all usages and the align with the base usage. Play song select is the only special one.